tests/pytests/unit/proxy/test_restconf.py

```diff
@@ -0,0 +1,89 @@
+import pytest
+import salt.proxy.restconf as restconf
+from tests.support.mock import patch
+
+
+@pytest.fixture(autouse=True)
+def setup_loader():
+    setup_loader_modules = {restconf: {}}
+    with pytest.helpers.loader_mock(setup_loader_modules) as loader_mock:
+        yield loader_mock
+
+
+@pytest.fixture
+def patch_conn_args():
+    with patch.dict(
+        restconf.restconf_device,
+        {
+            "conn_args": {
+                "hostname": None,
+                "transport": None,
+                "verify": None,
+                "username": None,
+                "password": None,
+            }
+        },
+    ):
+        yield
+
+
+@pytest.fixture
+def fake_query():
+    with patch("salt.utils.http.query", autospec=True) as fake_query:
+        yield fake_query
+
+
+def test_result_of_request_should_be_returned(patch_conn_args, fake_query):
+    expected_result = object()
+    fake_query.return_value = expected_result
+    result = restconf.request("https://example.com")
+    assert result is expected_result
+
+
+def test_if_None_is_provided_as_dict_payload_then_empty_string_should_be_provided_instead(
+    patch_conn_args, fake_query
+):
+    expected_data = ""
+    restconf.request("https://example.com", dict_payload=None)
+    call = fake_query.mock_calls[0]
+
+    assert call.kwargs["data"] == expected_data
+
+
+def test_if_text_is_provided_as_dict_payload_then_provided_string_should_be_used(
+    patch_conn_args, fake_query
+):
+    expected_data = "this is my fake payload"
+    restconf.request("https://example.com", dict_payload=expected_data)
+    call = fake_query.mock_calls[0]
+
+    assert call.kwargs["data"] == expected_data
+
+
```

Comment:
suggest something along the line of the following to skip Debian 9 where Py 3.5 is used and no sorted dictionary by default. Should resolve the failing Debian 9 tests
```
 33 @pytest.mark.skipif(
 34     sys.version_info[0] == 3 and sys.version_info[1] <= 5,
 35     reason="run on Python 3.6 or greater where OrderedDict is default",
 36 )
```

Reply:
This brings up an interesting question. Should we just fully fail loading of this module on anything less than 3.6. 

Reason being that incorrect changes could be sent to a switch if the order changes. 

For example on cisco switches, if your trying to set the order of allowed auth types on an interface, you need to send ordered data. ( i specifically remember this as one of the reasons for this check and how the order mattered )

Or is it just user beware...

Reply:
Debian 9 EOL's 30 June 2022, but it is 'old old stable' at this point and that is the last system with Python 3.5, which is EOL'd for sometime, even Python 3.6 EOL'd just before Christmas past.  But you raise a good point, and I am inclined to be user beware, since Debian 9 user's should have migrated to Debian 10 at least by now.  Understand there still Ubuntu 16.04 users which Salt no longer supports.

Could possibly add text stating **for use with Python 3.6 and above**, and then at least they were told.  Otherwise you have a lot of 'sorted' to add all because of a platform that is close to death, and a version of Python long dead.  I am inclined to add the text especially since 4+ months to go.

Reply:
Actually with 4+ months to go, I don't know if you even need that text.